### PR TITLE
Demo mode

### PR DESCRIPTION
**Pull request by hugokallstrom**

### CODE REVIEW

**Comment by hugokallstrom** [comment]
Will add DemoManager in a separate PR, and merge into this before merging this to dev 👍

**Review by StylianosGakis — APPROVED** [review]

File: app/core/core-demo-mode/src/main/kotlin/com/hedvig/android/core/demomode/DemoManager.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  override suspend fun setDemoMode(demoMode: Boolean) {
+    dataStore.edit {
+      it[demoModeKey] = demoMode
+    }
+  }
+
+  companion object {
+    private val demoModeKey = booleanPreferencesKey("demo-mode")
+  }
+}
```
[the remark this answers is not in the document]
Comment (hugokallstrom) [reply app/core/core-demo-mode/src/main/kotlin/com/hedvig/android/core/demomode/DemoManager.kt:34]: Good idea!

File: app/core/core-demo-mode/src/main/kotlin/com/hedvig/android/core/demomode/DemoManager.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  override suspend fun setDemoMode(demoMode: Boolean) {
+    dataStore.edit {
+      it[demoModeKey] = demoMode
+    }
+  }
+
+  companion object {
+    private val demoModeKey = booleanPreferencesKey("demo-mode")
```
[the remark this answers is not in the document]
Comment (hugokallstrom) [reply app/core/core-demo-mode/src/main/kotlin/com/hedvig/android/core/demomode/DemoManager.kt:32]: Great idea. I actually have encountered a bug just like you describe here, it was... not a cheap fix...

File: app/core/core-demo-mode/src/main/kotlin/com/hedvig/android/core/demomode/ProdOrDemoProvider.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
+package com.hedvig.android.core.demomode
+
+abstract class ProdOrDemoProvider<T>(
```
[the remark this answers is not in the document]
Comment (hugokallstrom) [reply app/core/core-demo-mode/src/main/kotlin/com/hedvig/android/core/demomode/ProdOrDemoProvider.kt:3]: Yes that makes sense

**Pull request merged**